src/cmd/new.rs

```diff
@@ -27,6 +27,10 @@ pub(crate) fn execute<P>(name: &str, dir: Option<P>) -> Result<Option<String>>
 where
     P: AsRef<Path>,
 {
+    if name.contains('.') {
```

Comment:
Rather than checking for individual invalid characters, we should rather only allow valid characters e.g. alphanumeric, underscores.